pkg/trace/api/profiles.go

```diff
@@ -22,48 +25,74 @@ const (
+func profilingEndpoints(apiKey string) []*traceconfig.Endpoint {
+	var endpoints []*traceconfig.Endpoint
+	main := profilingURLDefault
 	if v := config.Datadog.GetString("apm_config.profiling_dd_url"); v != "" {
-		return v
+		main = v
+	} else if site := config.Datadog.GetString("site"); site != "" {
+		main = fmt.Sprintf(profilingURLTemplate, site)
 	}
-	if site := config.Datadog.GetString("site"); site != "" {
-		return fmt.Sprintf(profilingURLTemplate, site)
+	endpoints = append(endpoints, &traceconfig.Endpoint{Host: main, APIKey: apiKey})
+
+	if opt := "apm_config.profiling_additional_endpoints"; config.Datadog.IsSet(opt) {
+		extra := config.Datadog.GetStringMapStringSlice(opt)
+		for endpoint, keys := range extra {
+			for _, key := range keys {
+				endpoints = append(endpoints, &traceconfig.Endpoint{Host: endpoint, APIKey: key})
+			}
+		}
 	}
-	return profilingURLDefault
+	return endpoints
 }
 
-// profileProxyHandler returns a new HTTP handler which will proxy requests to the profiling intake.
-// If the URL can not be computed because of a malformed 'site' config, the returned handler will always
+// profileProxyHandler returns a new HTTP handler which will proxy requests to the profiling intakes.
+// If the main intake URL can not be computed because of config, the returned handler will always
 // return http.StatusInternalServerError along with a clarification.
 func (r *HTTPReceiver) profileProxyHandler() http.Handler {
-	target := profilingEndpoint()
-	u, err := url.Parse(target)
-	if err != nil {
-		log.Errorf("Profile forwarder is OFF because of invalid intake URL: %v", err)
-		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-			msg := fmt.Sprintf("Agent is misconfigured with an invalid intake URL: %q", target)
-			http.Error(w, msg, http.StatusInternalServerError)
-		})
+	endpoints := profilingEndpoints(r.conf.APIKey())
+	var (
+		targets []*url.URL
+		keys    []string
+	)
+	for i, endpoint := range endpoints {
+		target, err := url.Parse(endpoint.Host)
+		if err != nil {
+			log.Errorf("Error parsing profiling intake URL %s: %v", endpoint.Host, err)
+			if i == 0 {
+				// when the main intake is incorrect we don't use additional
+				return errorHandler(endpoint.Host)
+			}
+			continue
+		}
+		targets = append(targets, target)
+		keys = append(keys, endpoint.APIKey)
 	}
 	tags := fmt.Sprintf("host:%s,default_env:%s", r.conf.Hostname, r.conf.DefaultEnv)
-	return newProfileProxy(r.conf.NewHTTPTransport(), u, r.conf.APIKey(), tags)
+	return newProfileProxy(r.conf.NewHTTPTransport(), targets, keys, tags)
+}
+
+func errorHandler(endpoint string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		msg := fmt.Sprintf("Profile forwarder is OFF because of invalid intake URL configuration: %v", endpoint)
+		http.Error(w, msg, http.StatusInternalServerError)
+	})
 }
 
 // newProfileProxy creates a single-host reverse proxy with the given target, attaching
```

Comment:
Let's correct this comment and explain how the index matches one-to-one between `targets` and `keys`, maybe something like:
```go
// newProfileProxy creates a new http.ReverseProxy which forwards incoming 
// requests to every target URL in targets, applying the API key at the same index
// from keys, using the given transport. The set of tags will be added as a header.
//
// For more details, see multiTransport.
```
I'm sure it can be phrased better :)